samples/algorithms/sudoku-grover/ColoringGroverWithConstraints.qs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.Quantum.Samples.ColoringGroverWithConstraints {
+
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Measurement;
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Diagnostics;
+
+    // Read color from a register
+    operation MeasureColor (register : Qubit[]) : Int {
+        return ResultArrayAsInt(MultiM(register));
+    }
+
+    // Read coloring from a register
+    operation MeasureColoring (K : Int, register : Qubit[]) : Int[] {
+        let V = Length(register) / K;
+        let colorPartitions = Partitioned(ConstantArray(V-1, K), register);
+        return ForEach(MeasureColor, colorPartitions);
+    } 
+
+    // N-bit color equality oracle (no extra qubits)
+    operation ColorEqualityOracle_Nbit (c0 : Qubit[], c1 : Qubit[], target : Qubit) : Unit is Adj+Ctl {
+        within {
+            for ((q0, q1) in Zip(c0, c1)) {
+                // compute XOR of q0 and q1 in place (storing it in q1)
+                CNOT(q0, q1);
+            }
+        } apply {
+            // if all XORs are 0, the bit strings are equal
+            (ControlledOnInt(0, X))(c1, target);
+        }
+    }
+
+    // Oracle for verifying vertex coloring, including color constraints from non QuBit vertexes
+    operation VertexColoringOracle (V : Int, K: Int, edges : (Int, Int)[],  startingColorConstraints : (Int, Int)[], colorsRegister : Qubit[], target : Qubit) : Unit is Adj+Ctl {
```

Comment:
This would be a great place to add docs for our API reference! We typically add these kinds of extended docblocks for all new operations. Please have a look at our [API reference docs](https://docs.microsoft.com/en-us/quantum/resources/contributing/docs#contributing-to-the-api-references) for guidance.

Reply:
done